### DISCUSSION TITLE
[openssl] build failure

### DISCUSSION THREAD

**Discussion by vlovch01** [microsoft/vcpkg#21348] [Q&A]
**Host Environment**
- OS: OSX
- Compiler:  AppleClang 12.0.5.12050022

**To Reproduce**
`./vcpkg install x86-ios`

**Failure logs** 
Build works fine, but the binary architecture type is 
x86_64 in stead of i386

**Additional context**
![image](https://user-images.githubusercontent.com/5650073/137822113-f9afff96-4aee-4b57-9bec-496192a74063.png)

**Comment by JonLiu1993** [comment]
@dg0yt ,Could you please take a look?

**Comment by vlovch01** [comment]
**arm-android** also affected

**Comment by dg0yt** [comment]
> @dg0yt Could you please take a look?

@PhoebeHui No, not for ios, and not for the buildsystem quirks in `ports/openssl/unix/CMakeLists.txt`

**Comment by vlovch01** [comment]
how to fix this?
